test/functional/plugin/lsp/diagnostic_spec.lua

```diff
@@ -573,5 +573,19 @@ describe('vim.lsp.diagnostic', function()
       eq('related bad!', related_diagnostics[1].message)
       eq('spongebob', relatedPreviousResultId)
     end)
+
+    it('refreshes diagnostics on server-to-client request', function()
```

Comment:
The test case looks somewhat incomplete. Shouldn't it also assert that the diagnostics get refreshed?